sdk/cognitivelanguage/ai-language-conversations/test/public/analyze.spec.ts

```diff
@@ -0,0 +1,246 @@
+
+        //Assert intent
+        var luis_result = top_intent_object.result.prediction;
+        assert.equal(luis_result.topIntent, "Reserve");
+        assert.isAtLeast(luis_result.intents.length, 1);
+        assert.isAbove(luis_result.intents.Reserve.confidence, 0);
+        
+        //Assert entities
+        assert.isAtLeast(luis_result.entities.length, 1);
+
+        done();
+      }).catch((error) => {
+        assert.fail();
+      })
+    })
+
+    it("Test Orchestration App QnA Response", (done) => {
+      client.analyzeConversation(conv4).then((message) => {
+        //Assert prediction type
+        assert.equal(message.kind, "ConversationResult");
+        assert.exists(message.result.query);
+        assert.equal(message.result.prediction.projectKind, "Orchestration");
+
+        //Assert top matching project
+        assert.equal(message.result.prediction.topIntent, "ChitChat-QnA");
+        assert.exists(message.result.prediction.intents["ChitChat-QnA"]);
+        var top_intent_object = message.result.prediction.intents["ChitChat-QnA"];
+        assert.equal(top_intent_object.targetProjectKind, "QuestionAnswering");
+
+        //Assert intent
+        var qna_result = top_intent_object.result;
+        assert.exists(qna_result.answers[0].answer);
+        assert.isAbove(qna_result.answers[0].confidence, 0);
+
+        done();
+      }).catch((error) => {
+        assert.fail();
+      })
+    })
+
+    it("Test Conversation App PII transcript", (done) => {
+      client.beginConversationAnalysisAndWait(conv5).then((message) => {
+        //Assert main object
+        assert.equal(message.status, "succeeded");
+
+        //Assert task result
+        var task_result = message.tasks.items[0];
+        assert.equal(task_result.status, "succeeded");
+        assert.equal(task_result.kind, "conversationalPIIResults");
+
+        //Assert Conversation Result
+        var conversation_items = task_result.results.conversations[0].conversationItems;
+        assert.exists(conversation_items);
+        conversation_items.forEach((conversation) => {
+          assert.exists(conversation.redactedContent);
+          assert.exists(conversation.entities);
+        })
+
+        done();
+      }).catch((error) => {
+        assert.fail();
+      })
+    })
+
+    it("Test Conversation Summarization App", (done) => {
+      client.beginConversationAnalysisAndWait(conv6).then((message) => {
+        //Assert main object
+        assert.equal(message.status, "succeeded");
+
+        //Assert task result
+        var task_result = message.tasks.items[0];
+        assert.equal(task_result.status, "succeeded");
+        assert.equal(task_result.kind, "conversationalSummarizationResults");
+
+        //Assert Conversation Result
+        var conversation_result = task_result.results.conversations[0];
+        assert.exists(conversation_result.summaries);
+        
+        done();
+      }).catch((error) => {
+        assert.fail();
```

Comment:
There is no need to wrap every test in a try/catch.

Could you please convert those functions into async ones and await Promises inside them?